src/SNPN/Common/RegexMatchHelper.cs

```diff
@@ -0,0 +1,13 @@
+using System;
+using System.Text.RegularExpressions;
+
+namespace SNPN.Common
+{
+	public class RegexMatchHelper {
+
+        public static bool MatchWholeWord(string input, string matchText) {
```

Comment:
While testing, I was having issues with the keyword detection since it's basically space + word + space. So I tried to make this keyword detection a bit more robust. Basically it's:
- match starts with start of string OR whitespace OR comma OR period
- word matches and is escaped properly
- match ends with end of string OR whitespace OR comma OR period OR exclamation OR question mark

This probably misses a bunch of cases, but it should help with things.

Reply:
Alright, Ichanged it to the suggestion. Yeah I was trying to be very specific on what characters would match. I'm ok with going simpler here and if people complain about matches then we can look at this regex again.